powerbox/powerbox.py

```diff
@@ -247,14 +247,31 @@ def create_discrete_sample(self, nbar, randomise_in_cell=True, min_at_zero=False
             origin.
         store_pos : bool, optional
             Whether to store the sample of tracers as an instance variable `tracer_positions`.
+        delta_x : numpy.ndarray
+            Field from which to draw discrete samples. This is likely the
+            output of a previous call to `delta_x()`, but could in principle be
+            any field. Note that if not supplied, the field will be generated
+            from scratch. As a result, unless the user has supplied a random seed
+            at initialization, the discrete samples will be a new realization of
+            a field with the specified power spectrum.
 
         Returns
         -------
         tracer_positions : float, array_like
             ``(n, d)``-array, with ``n`` the number of tracers and ``d`` the number of dimensions. Each row represents
             a single tracer's co-ordinates.
         """
-        dx = self.delta_x()
+
+        if delta_x is None:
+            if self.seed is None:
+                warn = "WARNING: should provide `seed` at initialization if one"
+                warn += " wants a correspondence between parent field and"
+                warn += " discrete samples."
+                print(warn)
```

Comment:
Two things: 

1. Use `warnings.warn` instead of print
2. You can put the full message inside the call, like

```python
warnings.warn(
    "WARNING: should provide `seed` at initialization if one"
    " wants a correspondence between parent field and"
    " discrete samples."
)
```